lib/page/questions/widget/question_page_widget.dart

```diff
@@ -62,10 +64,10 @@ class QuestionsItemWidget extends StatelessWidget {
                       ? RoundedButtonWidget(
                           buttonText:
                               AppLocalizations.of(context)!.questionsSubmit,
-                          onPressed: () => onButtonPressed.call(),
+                          onPressed: () => onNextQuestion.call(),
                         )
                       : NextButtonWidget(
-                          onPressed: () => onButtonPressed.call(),
+                          onPressed: () => onSubmitSurvey.call(),
```

Comment:
Are you making a mistake here by calling wrong callbacks? Next button with `onSubmitSurvey` and Submit button with `onNextQuestion`